### ISSUE TITLE
Exception thrown: 'System.ApplicationException' in Umbraco.UIBuilder.Infrastructure.dll

### ISSUE THREAD

**Issue by matt-tolliday-iss** [umbraco/Umbraco.UIBuilder.Issues#94]
**Describe the bug**
We are seeing a Exception thrown: 'System.ApplicationException' in Umbraco.UIBuilder.Infrastructure.dll when attempting to read an entity from either the repository .Get or .GetAll methods. When using the .GetAll with a lambda, the error is: Exception has been thrown by the target of an invocation.

This code is being executed from within a Umbraco.UIBuilder.Configuration.Actions.Action<ActionResult>

   at Umbraco.UIBuilder.Persistence.DefaultRepositoryProxy`2.Get(TId id, Boolean raiseEvents)
   at XXX.XXX.Umbraco.Infrastructure.Register.XXX.Execute(String collectionAlias, Object[] entityIds) in 

There are no errors in the Umbraco Logs and there doesn't appear to be anything wrong with the configuration. 

The "Status" value was mapped to an enum, which was causing the same error as above under the EventMessages property of the GetAll operationresult (Exception has been thrown by the target of an invocation.)

Model:

```
[TableName("pbMemberMetaData")]
[PrimaryKey("Id", AutoIncrement = true)]
[ExplicitColumns]
public class MemberMetaDataDto
{
    public int Id { get; set; }
    
    public int MemberId { get; set; }

    public int Status { get; set; }
}
```

Database schema:

<img width="148" alt="image" src="https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/80101704/776f7b6b-c89c-4682-a8f3-cdfc4a4e51f6">

Repository code to call entity:

`var entity = repository.GetAll(x => x.MemberId == id);`

**Expected behavior**
The entity should be returned from the repository when found by expression.

**Screenshots**

**Environment (please complete the following information):**
 - Server OS: Windows 11
 - Browser Chrome
 - Umbraco Version 13.0.3
 - Product Version 13.0.2, 13.0.3, 13.1.0-rc2

**Comment by acoumb** [comment]
Hi @matt-tolliday-iss ,

I've tried replicating your configuration using the same table structure, POCO model and with this sample action, but could not replicate the issue:
```
public class EnquiryAction : UIBuilder.Configuration.Actions.Action<ActionResult>
{
    private readonly IRepositoryFactory _repositoryFactory;

    public EnquiryAction(IRepositoryFactory repositoryFactory) => _repositoryFactory = repositoryFactory;

    public override string Alias => "enquiry-action";

    public override string Name => "Enquiry Action";

    public override bool IsVisible(ActionVisibilityContext ctx)
    {
        return ctx.ActionType == ActionType.Bulk
            || ctx.ActionType == ActionType.Row;
    }

    public override ActionResult Execute(string collectionAlias, object[] entityIds)
    {
        try
        {
            var repo = _repositoryFactory.GetRepository<Enquiry, int>(collectionAlias);

            var result = repo.GetAll(x => x.MemberId == 1055);

            return new ActionResult(true);
        }
        catch (Exception ex)
        {
            return new ActionResult(false, new ActionNotification("Failed to update status", ex.Message));
        }
    }
}
```
Also called the repository successfully from a  test controller:
```
public class TestController : UmbracoApiController
{
    private readonly IRepositoryFactory _repositoryFactory;

    public TestController(IRepositoryFactory repositoryFactory) => _repositoryFactory = repositoryFactory;

    public IActionResult Index()
    {
        var repository = _repositoryFactory.GetRepository<Enquiry, int>();
        var list = repository.GetAll(x => x.MemberId == 1055);
        return new JsonResult(list);
    }
}
```

Could you provide more context that could help clarify and replicate the issue?

Thanks,
Adrian

**Comment by matt-tolliday-iss** [comment]
Thanks Adrian,

It's being registered like the following, so that it never technically is visible, but can be "used" with the repository code:

<img width="884" alt="image" src="https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/80101704/218223a6-32b0-4849-ac3e-718a66accbe7">

Usgae is then like:

<img width="669" alt="image" src="https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/80101704/537d8367-13d0-42d8-ad45-0d7b3602ca5b">

Variables bind, injected fine:

<img width="830" alt="image" src="https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/80101704/3e734409-b455-4ada-8100-be84bbef3a80">

That entity exists in database:

<img width="173" alt="image" src="https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/80101704/ab80991a-2548-4411-8819-bf937649bde0">


Which results in:

<img width="713" alt="image" src="https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/80101704/536edb80-9f03-4cea-9df5-d7b160b3991a">

**Comment by acoumb** [comment]
Hi @matt-tolliday-iss ,

I eventually managed to catch the exception with this `where` clause
```
var result = repo.GetAll(x => x.MemberId == int.Parse(member.Id));
```
and it was caused by using `Parse` in a lambda expression (`ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'No logic supported for Parse')`), so I would recommend not doing that.

This worked instead ```var result = repo.GetAll(x => x.MemberId.Equals(member.Id));```.

With your sample code it worked fine though.
![image](https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/95346674/8290c45c-5e29-4ff1-b18f-136ad300d9e6)
